android-interop-testing/src/generated/debug/grpc/io/grpc/testing/integration/MetricsServiceGrpc.java

```diff
@@ -199,6 +214,49 @@ public void getGauge(io.grpc.testing.integration.Metrics.GaugeRequest request,
   /**
    * A stub to allow clients to do synchronous rpc calls to service MetricsService.
    */
+  public static final class MetricsServiceBlockingV2Stub
+      extends io.grpc.stub.AbstractBlockingStub<MetricsServiceBlockingV2Stub> {
+    private MetricsServiceBlockingV2Stub(
+        io.grpc.Channel channel, io.grpc.CallOptions callOptions) {
+      super(channel, callOptions);
+    }
+
+    @java.lang.Override
+    protected MetricsServiceBlockingV2Stub build(
+        io.grpc.Channel channel, io.grpc.CallOptions callOptions) {
+      return new MetricsServiceBlockingV2Stub(channel, callOptions);
+    }
+
+    /**
+     * <pre>
+     * Returns the values of all the gauges that are currently being maintained by
+     * the service
+     * </pre>
+     */
+    public io.grpc.stub.BlockingClientCall<io.grpc.testing.integration.Metrics.EmptyMessage,io.grpc.testing.integration.Metrics.GaugeResponse>
```

Comment:
I thought that the request message had to be the type specified in the proto file.  Is that not true?  Having the type allows IDEs to generate matching code and the compiler is validating the argument.   Am I missing something here?

Reply:
The request type is only used in `blockingClientCall.write(ReqT)`. But they shouldn't call that method because the call has already been half-closed. So the request type won't actually be used.